unit-tests-java8/src/test/java/org/eclipse/collections/test/map/immutable/ImmutableUnifiedMapTest.java

```diff
@@ -40,6 +43,37 @@
         return result.toImmutable();
     }
 
+    @Override
+    public <K, V> ImmutableMap<K, V> newWithKeysValues(Object... elements)
+    {
+        if (elements.length % 2 != 0)
+        {
+            fail(String.valueOf(elements.length));
+        }
+
+        MutableMap<K, V> result = new UnifiedMap<>();
+        for (int i = 0; i < elements.length; i += 2)
+        {
+            assertNull(result.put((K) elements[i], (V) elements[i + 1]));
+        }
+        return result.toImmutable();
+    }
+
+    @Override
+    public void MapIterable_flipUniqueValues()
+    {
+        MapIterable<String, Integer> map = this.newWithKeysValues("Three", 3, "Two", 2, "One", 1);
+        MapIterable<Integer, String> result = map.flipUniqueValues();
+
+        // TODO: Use IterableTestCase.assertEquals instead, after setting up methods like getExpectedTransformed, but for maps.
+        Assert.assertEquals(
+                UnifiedMap.newWithKeysValues(3, "Three", 2, "Two", 1, "One"),
+                result);
+
+        // TODO: Fix bug, flipUniqueValues is implemented incorrectly in immutable maps. Compare ImmutableDoubletonMap and DoubletonMap for example.
+        this.newWithKeysValues(1, "2", 2, "2").flipUniqueValues();
```

Comment:
This is a real bug. Calling flipUniqueValues() on a map with duplicate values is supposed to throw. It fails for immutable fixed size maps. When we fix it, we also need to add coverage here. This one only catches Doubleton.

Reply:
This I presume we will have to fix as a part of major release as it might break existing users who do not have exception handling for duplicates, or can we fix it in a minor release?

Reply:
```java
    /**
     * Return the MapIterable that is obtained by flipping the direction of this map and making the associations from value to key.
     * @throws IllegalStateException if the MapIterable contains duplicate values.
     * @since 5.0
     */
    MapIterable<V, K> flipUniqueValues();
```

Given that the implementation violates its own contract, I think its fine to include in a bug-fix release if that's how you want to proceed.

It's an old bug that's been there since the first implementation of flipUniqueValues, so it's not urgent either.

Reply:
10.0.0 will open in May/June I would rather wait on the bug fix till then.

Reply:
Raised #443